tests/Maintenance/MaintenanceTest.cpp

```diff
@@ -0,0 +1,527 @@
+    size_t one = 1;
+    std::string hello("hello world!");
+    auto props = std::make_shared<VPackBuilder>();
+    { VPackObjectBuilder obj(props.get());
+      props->add(VPackValue("array"));
+      { VPackArrayBuilder arr(props.get());
+        props->add(VPackValue(pi));
+        props->add(VPackValue(one));
+        props->add(VPackValue(hello)); }}
+    ActionDescription desc({{"name", "SomeAction"}}, props);
+    REQUIRE(desc.get("name") == "SomeAction");
+    REQUIRE(desc.properties()->slice().hasKey("array"));
+    REQUIRE(desc.properties()->slice().get("array").isArray());
+    REQUIRE(desc.properties()->slice().get("array").length() == 3);
+    REQUIRE(desc.properties()->slice().get("array")[0].getNumber<double>()==pi);
+    REQUIRE(desc.properties()->slice().get("array")[1].getNumber<size_t>()==one);
+    REQUIRE(desc.properties()->slice().get("array")[2].copyString() == hello );
+  }
+    
+}
+
+TEST_CASE("ActionPhases", "[cluster][maintenance]") {
+
+  auto plan = createNode(planStr);
+  auto current = createNode(currentStr);
+
+  std::vector<std::string> dbsIds;
+  for (auto const& dbs : plan("Plan/DBServers").children()) {
+    dbsIds.push_back(dbs.first);
+  }
+  
+  std::map<std::string, Node> localNodes {
+    {dbsIds[2], createNode(dbs1Str)},
+    {dbsIds[0], createNode(dbs2Str)},
+    {dbsIds[1], createNode(dbs3Str)}};  
+
+  SECTION("In sync should have 0 effects") {
+    
+    std::vector<ActionDescription> actions;
+    
+    for (auto const& node : localNodes) {
+      arangodb::maintenance::diffPlanLocal(
+        plan.toBuilder().slice(), node.second.toBuilder().slice(),
+        node.first, actions);
+
+      if (actions.size() != 0) {
+        //std::cout << actions << std::endl;
+      }
+      REQUIRE(actions.size() == 0);
+    }
+
+  }
+
+
+  SECTION("Local databases one more empty database should be dropped") {
+
+    std::vector<ActionDescription> actions;
+    localNodes.begin()->second("db3") =
+      arangodb::velocypack::Slice::emptyObjectSlice();
+    
+    arangodb::maintenance::diffPlanLocal(
+      plan.toBuilder().slice(), localNodes.begin()->second.toBuilder().slice(),
+      localNodes.begin()->first, actions);
+
+    REQUIRE(actions.size() == 1);
+    REQUIRE(actions.front().name() == "DropDatabase");
+    REQUIRE(actions.front().get("database") == "db3");
+
+  }
+
+
+  SECTION("Local databases one more non empty database should be dropped") {
+
+    std::vector<ActionDescription> actions;
+    localNodes.begin()->second("db3/col") =
+      arangodb::velocypack::Slice::emptyObjectSlice();
+    
+    arangodb::maintenance::diffPlanLocal(
+      plan.toBuilder().slice(), localNodes.begin()->second.toBuilder().slice(),
+      localNodes.begin()->first, actions);
+
+    REQUIRE(actions.size() == 1);
+    REQUIRE(actions.front().name() == "DropDatabase");
+    REQUIRE(actions.front().get("database") == "db3");
+
+  }
+
+  
+  SECTION("Local databases one more empty database should be dropped") {
```

Comment:
redundant to two tests up.  was this supposed to be a shard?

Reply:
fine the way it is